bin/orcabus.ts

```diff
@@ -14,6 +14,12 @@ const props: cdk.StackProps = {
   },
 };
 
-new OrcaBusStatefulStack(app, 'OrcaBusStatefulStack', { ...props, ...orcaBusStatefulConfig });
+const orcaBusStatefulStack = new OrcaBusStatefulStack(app, 'OrcaBusStatefulStack', {
+  ...props,
+  ...orcaBusStatefulConfig,
+});
 
-new OrcaBusStatelessStack(app, 'OrcaBusStatelessStack', { ...props, ...orcaBusStatelessConfig });
```

Comment:
Before changing more on this, instead of passing props of string (e.g. eventBus name, SG name) I think we should just pass the resource itself, so it doesn't need for a lookup anymore. (As it is on the example of [Stateless vs Stateful](https://blog.serverlessadvocate.com/serverless-aws-cdk-pipeline-best-practices-patterns-part-1-ab80962f109d) blog)

Or was this is done on purpose ?

Reply:
Ah. That blog post. 😅 It is on purpose, yes, Will. Because we do not want to pass around "resource reference". Considering situation when some consuming Construct get destroy (i.e. Stateless/self-mutating part), we do not want to either get deleted stateful resource by accident nor get entangled not able to destroy Stateless Construct. So, falling this back to the lowest primitive possible as config string or number.